### DISCUSSION TITLE
PIR with screen switching does not behave as expected

### DISCUSSION THREAD

**Discussion by catdogmaus** [arendst/Tasmota#22293] [Q&A]
I attempted to use pir similar to AM312 to switch screen on and off using this.

```
rule3
 on tele-DS18B20-1#Temperature do DisplayText [l2c1f3s2]Boiler %value% C~20~20~20 endon
 on tele-DS18B20-2#Temperature do DisplayText [l4c1f3s2]Stove %value% C~20~20~20 endon
 on Switch3#state=2 do Backlog Power3 1; RuleTimer1 30 endon 
 on Rules#Timer=1 do backlog Power3 0 endon
```

However it seems to have some errors bcs it does not work as expected. 
It _does_ switch screen on and off, but 30sec delay does apply not only _off_ but also to _on_ switch. That means to see the screen I have to wait 30sec and after additional 30sec it then switches off. Obviously the desired behavior should be instant _on_ as all pir sensors do.
thanks

**Comment by sfromis** [comment]
When using a [Switch](https://tasmota.github.io/docs/Buttons-and-Switches/#switch) input, you **must** select a `SwitchMode` to match your desired behavior, and the default of 0 for toggle is doing what you post about.

Instead of that RuleTimer and last rule, you could also do `TimedPower3 30000,on` which will automatically go back to off after the interval in ms.

**Reply by catdogmaus** [reply]
I actually still do not understand fully what `SwitchMode` is proper to use with pir. In example I used above was from here https://tasmota.github.io/docs/Project-AM312-and-Sonoff-R2/ and there is told to use `SwitchMode 4`. when I think about pir it should work more as `SwitchMode 1` - so more like regular switch. Signal from pir - ON, no signal - OFF. 

Also I have now issue that after restart the screen is always ON. The reason to use pir for screen is to keep it off by default. `PowerOnState 0` has no effect about that behavior. 

Additional question not about pir but about the same display, I use ST7735 display and configured BL pin as `Backlight`. So far there seems to be no difference to actually power it directly from VCC. I also use Universal Display Driver.
Is it actually possible to turn backlight off together with display?

**Reply by sfromis** [reply]
The "problem" with `SwitchMode 1` could be if the timing provided by the PIR matches, or pauses of detection might lead to it switching off too early. It is common that a PIR has some delay before it signals no activity, but that timing may or may not be what you'd like.

That's why the page you linked is suggesting `SwitchMode 4`, along with a rule to have the output turned on every time some activity was detected, and then automatically off after a set interval, where the example has 180 seconds.

Whether this matches what you'd like is of course another question, but what I posted is the reasoning behind the suggestion, like in case of using it for a bathroom fan, where there can be "good reasons" to ensure it runs for a bit of extra time. 

Short version is that if you get to a setup which works for you, that's "proper".

If you want it off after restarts, you could have boot time rules to ensure your desired behavior. Using `PowerOnState` is *literally* for power-on, not for restarts.

With displays, you can usually do a power on/off (including on the front page) to select it to be lit or not, where I see the backlight pin working for this.

**Reply by catdogmaus** [reply]
> 

> With displays, you can usually do a power on/off (including on the front page) to select it to be lit or not, where I see the backlight pin working for this.

Thanks. 
It would be great (and in my mind logical) when, if backlight pin configured it would turn ON/OFF with screen by default. If there is somebody who does not want that option, for that case it can be always connected directly to VCC.

**Reply by sfromis** [reply]
In many cases, there is no "it" which "can always be connected to directly to VCC", as how on/off is managed depends on the display type, including on displays controlled via I2C without other pins involved.

I see it as "plausibly useful" to separate the control of brightness from on/off, making it easier to have automations simply turning it on/off, while the brightness is handled by other means, maybe depending on ambient light. Like on a radio where you can have a set volume not to be readjusted every time you turn it on/off.

**Reply by catdogmaus** [reply]
> I see it as "plausibly useful" to separate the control of brightness from on/off, making it easier to have automations simply turning it on/off, while the brightness is handled by other means, maybe depending on ambient light.

It should be probably possible to make different options with backlight pin (1, 2 or 3) binding those with different behavior. With rules can probably be done most of things but with "out of the box" options will always give better user experience.
